src/pytorch_lightning/core/saving.py

```diff
@@ -206,11 +210,11 @@ def _load_from_checkpoint(
 
 
 def _load_state(
-    cls: Union["pl.LightningModule", "pl.LightningDataModule"],
+    cls: Union[Type["ModelIO"], Type["pl.LightningModule"], Type["pl.LightningDataModule"]],
     checkpoint: Dict[str, Any],
     strict: Optional[bool] = None,
     **cls_kwargs_new: Any,
-) -> Any:
+) -> Union["ModelIO", "pl.LightningModule", "pl.LightningDataModule"]:
```

Comment:
this should never return ModelIO however, the implementation of `_load_from_checkpoint` in core.saving is such that this return type is forced to appease mypy (based on my understanding).